File: app/src/main/java/com/gb/rating/models/utils/PointRepository.java
package com.gb.rating.models.utils;

import org.jetbrains.annotations.NotNull;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Maybe;
import io.reactivex.annotations.NonNull;

public interface PointRepository<GenericPoint> {

    @NonNull
    Completable writePoint(@NotNull GenericPoint value);

    @NonNull
    Maybe<List<GenericPoint>> retrievePoints(@NotNull String country, @NotNull String city, String googleType, double latitudeFrom, double latitudeTo, double longitudeFrom, double longitudeTo, String deleted);

    @NonNull
    Maybe<List<GenericPoint>> retrieveNewPoints(@NotNull String country, @NotNull String city, @NotNull String cafeGoogleType, long changeTime);

    long[] writePoints(List<GenericPoint> points);
}

File: app/src/main/java/com/gb/rating/models/repository/PointRepository.kt
package com.gb.rating.models.repository

import com.gb.rating.filestore.Point_FB
import com.gb.rating.models.CafeItem
import io.reactivex.Maybe
import io.reactivex.annotations.NonNull

interface PointRepository {

    @NonNull
    fun writePoint(@NonNull value: Point_FB): io.reactivex.Completable

    @NonNull
    fun retrievePoints(@NonNull country: String, @NonNull city: String, googleType: String, latitudeFrom: Double, latitudeTo: Double, longitudeFrom: Double, longitudeTo: Double): Maybe<List<Point_FB>>

}

